### PR TITLE
OO-sunu #11 adding the paymentType to kunde and KontoDaten parameters…

### PR DESCRIPTION
**Pull request by mcmpp**
… as well as linking KontoDaten to kunde

### CODE REVIEW

File: src/main/scala/ch/openolitor/stammdaten/StammdatenUpdateService.scala [hunk 1]
```diff
@@ -205,6 +205,7 @@ class StammdatenUpdateService(override val sysConfig: SystemConfig) extends Even
       DB localTxPostPublish { implicit session => implicit publisher =>
         updateKundendaten(meta, kundeId, update)
         updatePersonen(meta, kundeId, update)
+        updateKonteDaten(meta, kundeId, update)
```
Comment (AlwinEgger) [inline src/main/scala/ch/openolitor/stammdaten/StammdatenUpdateService.scala]: Typo: It's "updateKont**o**Daten"

File: src/main/scala/ch/openolitor/stammdaten/models/KontoDaten.scala [hunk 2]
```diff
@@ -8,9 +8,13 @@ case class KontoDatenId(id: Long) extends BaseId
 
 case class KontoDaten(
   id: KontoDatenId,
+  nameAccountHolder: Option[String],
+  addressAccountHolder: Option[String],
+  bankName: Option[String],
   iban: Option[String],
   teilnehmerNummer: Option[String],
   referenzNummerPrefix: Option[String],
+  kunde: Option[KundeId],
```
Comment (AlwinEgger) [inline src/main/scala/ch/openolitor/stammdaten/models/KontoDaten.scala]: This means that the one record with "kunde" -> None is the entry for the project?

File: src/main/scala/ch/openolitor/stammdaten/StammdatenUpdateService.scala [hunk 3]
```diff
@@ -235,6 +236,21 @@ class StammdatenUpdateService(override val sysConfig: SystemConfig) extends Even
     }
   }
 
+  private def updateKonteDaten(meta: EventMetadata, kundeId: KundeId, update: KundeModify)(implicit session: DBSession, publisher: EventPublisher, personId: PersonId = meta.originator): Unit = {
+    val kontoDaten = stammdatenWriteRepository.getKontoDaten(kundeId)
```
Comment (AlwinEgger) [inline src/main/scala/ch/openolitor/stammdaten/StammdatenUpdateService.scala]: As there should always be only one entry KontoDaten per Kunde, getKontoDaten should not return a list.
But: Wouldn't it be more coherent to connect the entities the other way around?

File: src/main/scala/ch/openolitor/stammdaten/models/KontoDaten.scala (hunk 2 above)
Comment (mcmpp) [reply src/main/scala/ch/openolitor/stammdaten/models/KontoDaten.scala]: yes

**Pull request closed without merging**